src/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv/Internal/LibuvConnection.cs

```diff
@@ -86,29 +83,13 @@ public LibuvConnection()
                     // Ensure the socket is disposed prior to completing in the input writer.
                     _socket.Dispose();
                     Input.Complete(new TaskCanceledException("The request was aborted"));
-                    _socketClosedTcs.TrySetResult(null);
+                    _connectionContext.OnConnectionClosed();
```

Comment:
The code makes it look like there's a chance that `OnConnection` can throw and as a result `OnConnectionClosed` won't be called.